src/Tribe/Integrations/WPML/Views/V2/Filters.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Manages integration between WPML and Views v2 of The Events Calendar.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Integrations\WPML\Views\V2
+ */
+
+namespace Tribe\Events\Integrations\WPML\Views\V2;
+
+/**
+ * Class Filters
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Integrations\WPML\Views\V2
+ */
+class Filters {
+	/**
+	 * Translates the View URL.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $url The original View URL.
+	 *
+	 * @return string The translated View URL.
+	 */
+	public static function translate_view_url( $url ) {
+		$lang = static::get_request_lang();
+		if ( false === $lang ) {
+			return $url;
+		}
+
+		return apply_filters( 'wpml_permalink', $url, $lang );
+	}
+
+	/**
+	 * Returns the current request language, read from the request cookie.
+	 *
+	 * @since TBD
+	 *
+	 * @return string|false Either the request language, e.g. `fr`, or `false` to indicate the language could not be
+	 *                      parsed from the request context or that the request context is the current one and there
+	 *                      is no need to translate URLs.
+	 */
+	protected static function get_request_lang() {
+		if ( ! (
+			function_exists( 'icl_object_id' )
+			&& isset( $_COOKIE['wp-wpml_current_language'] )
+		) ) {
```

Comment:
:no_entry_sign: Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.

<sub>hash: 748c73929e2d4b6d0189702ae9e645f1</sub>